truss/main.go

```diff
@@ -79,82 +79,118 @@ func parseInput() (*truss.Config, error) {
 		return nil, errors.Wrap(err, "cannot parse input arguments")
 	}
 
-	// PBGoPackage
-	if *pbPackageFlag == "" {
-		return &cfg, nil
+	// Service Path
+	defFile, err := os.Open(cfg.DefPaths[0])
+	if err != nil {
+		return nil, errors.Wrapf(err, "Could not open package file %q", cfg.DefPaths[0])
+	}
+	svcName, err := parsepkgname.PackageNameFromFile(defFile)
+	if err != nil {
+		return nil, errors.Wrapf(err, "cannot parse package name from file %q", cfg.DefPaths[0])
+	}
+	svcFolderName := svcName + "-service"
+	svcPath := filepath.Join(filepath.Dir(cfg.DefPaths[0]), svcFolderName)
+	cfg.ServicePackage, err = filepath.Rel(filepath.Join(cfg.GOPATH, "src"), svcPath)
+	if err != nil {
+		return nil, errors.Wrap(err, "service path is not in GOPATH")
 	}
 
-	cfg.PBPackage = *pbPackageFlag
-	if !fileExists(
-		filepath.Join(cfg.GOPATH, "src", cfg.PBPackage)) {
-		return nil, errors.Errorf(".pb.go output package directory does not exist: %q", cfg.PBPackage)
+	// PrevGen
+	cfg.PrevGen, err = readPreviousGeneration(cfg.ServicePath())
+	if err != nil {
+		return nil, errors.Wrap(err, "cannot read previously generated files")
+	}
+
+	// PBGoPackage
+	if *pbPackageFlag == "" {
+		cfg.PBPackage = cfg.ServicePackage
+	} else {
+		cfg.PBPackage = *pbPackageFlag
+		if !fileExists(
+			filepath.Join(cfg.GOPATH, "src", cfg.PBPackage)) {
+			return nil, errors.Errorf(".pb.go output package directory does not exist: %q", cfg.PBPackage)
+		}
 	}
 
 	return &cfg, nil
 }
 
-// parseServiceDefinition returns a deftree which contains all needed for all
-// generating a truss service and documentation
-func parseServiceDefinition(definitionPaths []string) (deftree.Deftree, error) {
-	protocOut, err := execprotoc.CodeGeneratorRequest(definitionPaths)
-	if err != nil {
-		return nil, errors.Wrap(err, "cannot use parse input files with protoc")
+// parseServiceDefinition returns a deftree which contains all necessary
+// information for generating a truss service and its documentation.
+func parseServiceDefinition(cfg *truss.Config) (deftree.Deftree, *svcdef.Svcdef, error) {
```

Comment:
Is the thinking here that eventually this will just return a `Svcdef`? Is the `Deftree` here just for compatibility with the current state of Truss? Am I understanding your intent correctly?